src/test_l3_tvSettings.c

```diff
@@ -0,0 +1,5224 @@
+    result = SaveSourcePictureMode(videoSource, videoFormat, selectedPictureMode);
+    UT_LOG_INFO("Result SaveSourcePictureMode(IN:VideoSrcType:[%s], IN:VideoFormatType:[%s], IN:PictureMode:[0x%0X]) tvError_t:[%s]",
+                 ut_control_GetMapString(tvVideoSrcType_mapTable, videoSource),
+                 ut_control_GetMapString(tvVideoFormatType_mapTable, videoFormat),
+                 selectedPictureMode,
+                 ut_control_GetMapString(tvError_mapTable, result));
+
+    ASSERT(result == tvERROR_NONE);
+
+    UT_LOG_INFO("Out %s", __FUNCTION__);
+}
+
+/**
+ * @brief Saves the CMS value.
+ *
+ * This function saves the CMS value in the override picture profile database for the specific picture mode, primary video format type
+ * and primary video source. The saved CMS value should automatically take effect whenever the
+ * specified picture mode is selected, specified primary video format is played, specified primary video source is selected
+ * and the CMS state is enabled for that combination. There will be no change in the current CMS value applied in the color management system.
+ * When the component_type is passed as COMP_NONE and color_type is passed as tvDataColor_NONE, the cms_value
+ * refers to CMS state.
+ *
+ * **Test Group ID:** 04
+ * **Test Case ID:** 60
+ *
+ * **Test Procedure:**
+ * Refer to Test specification documentation
+ * [tvSettings_L3_CMS_TestSpecification.md](../docs/pages/tvSettings_L3_CMS_TestSpecification.md)
+ */
+void test_l3_tvSettings_CMSSave(void)
+{
+    // Declare and initialize all variables at the top
+    gTestID = 60;
+    UT_LOG_INFO("In %s [%02d%03d]", __FUNCTION__, gTestGroup, gTestID);
+
+    tvError_t result = tvERROR_NONE;
+    tvVideoSrcType_t videoSource = VIDEO_SOURCE_IP; // Set video source to IP directly
+    tvVideoFormatType_t videoFormat = VIDEO_FORMAT_NONE;
+    int cmsValue = 0;
+    tvComponentType_t componentType = COMP_NONE;
+    tvDataComponentColor_t colorType = tvDataColor_NONE;
+    int pqMode = 0;
+    int userColorChoice;
+    int maxRange = 30;
+    int menuIndex = 1;
+
+    UT_LOG_MENU_INFO("Using Video Source: VIDEO_SOURCE_IP");
+
+    // List all options for Picture Modes
+    UT_LOG_MENU_INFO("----------------------------------------------------------");
+    UT_LOG_MENU_INFO("\t\tSupported Picture Modes");
+    UT_LOG_MENU_INFO("----------------------------------------------------------");
+    UT_LOG_MENU_INFO("\t#   %-30s", "Picture Modes");
+    for (unsigned int k = PQ_MODE_STANDARD; k <= PQ_MODE_VIVID2; k++)
+    {
+        UT_LOG_MENU_INFO("[%u]: [%s]", k + 1, ut_control_GetMapString(tvPQModeIndex_mapTable, k)); // Display index starting from 1
+    }
+    UT_LOG_MENU_INFO("----------------------------------------------------------");
+    UT_LOG_MENU_INFO("Enter your choice of Picture Mode (index): ");
+    readInt(&pqMode);
+
+    // Validate picture mode choice
+    if (pqMode < 1 || pqMode > (int)PQ_MODE_VIVID2)
```

Comment:
use enum values instead of 1, check other places

Reply:
Done